### PR TITLE
Python 2.7 support for Python postinstall script

### PR DESCRIPTION
**Pull request by zrhoffman**
## What does this PR (Pull Request) do?

* This PR makes the Python postinstall script added in #4763 compatible with Python 2.7.
* This PR also makes the Python postinstall script compatible with `/usr/lib/platform-python` (Python 3.6) on CentOS 8, which does not have the `typing` module.

- [x] This PR is not related to any Issue 

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?

* Run the postinstall end-to-end tests against Python 2.7 on CentOS 7 after installing the test dependencies:
  ```shell
  <<'COMMANDS' docker run --rm -i -v $(pwd):/trafficcontrol -w /trafficcontrol centos:7 bash
      yum -y install epel-release file openssl &&
      yum -y install jq &&
      time traffic_ops/install/bin/postinstall.test.sh -2 &&
      echo Tests succeeded for CentOS 7
  COMMANDS
  ```

* Run the postinstall end-to-end tests against Python 3.6 on CentOS 8 after installing the test dependencies:
  ```shell
  <<'COMMANDS' docker run --rm -i -v $(pwd):/trafficcontrol -w /trafficcontrol centos:8 bash
  yum -y install epel-release file openssl &&
  yum -y install jq &&
  time traffic_ops/install/bin/postinstall.test.sh -3 -s -b /usr/libexec/platform-python &&
  echo Tests succeeded for CentOS 8
  COMMANDS
  ```

## The following criteria are ALL met by this PR

- [x] This PR includes tests
- [x] This PR includes documentation
- [x] An update to CHANGELOG.md is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_ops/install/bin/postinstall.py [hunk 1]
```diff
@@ -560,48 +703,52 @@ def unmarshal_config(dct: dict) -> Dict[str, List[Question]]:
...
-	if maxmind_answer.casefold() not in {'y', 'yes'}:
+	if maxmind_answer.lower() not in {'y', 'yes'}:
 		logging.info("Not downloading Maxmind data")
 		return
 
 	os.chdir(os.path.join(root, 'opt/traffic_ops/app/public/routing'))
 
+	def failed_download(ip_version):  # type: (int) -> None
+		logging.error("Failed to download MaxMind data")
+		logging.debug("(ipv%d) Exception: %s", ip_version, e)
```
Comment (ocket8888) [inline traffic_ops/install/bin/postinstall.py:719]: `e` is not defined in the scope of this function

File: traffic_ops/install/bin/postinstall.py [hunk 2]
```diff
@@ -1167,34 +1338,41 @@ def main(
...
+				"Use the script `/opt/traffic_ops/install/bin/todb_bootstrap.sh` "
 				"on the db server to create it and run `postinstall` again."
 			)
+
+		try:
+			setup_database_data(conn_str, admin_conf, paramconf, root_dir)
+		except (OSError, subprocess.CalledProcessError)as e:
+			db_connect_failed()
+			return -1
+		except subprocess.SubprocessError as e:
```
Comment (ocket8888) [inline traffic_ops/install/bin/postinstall.py:1354]: Why split these `excepts` if they have the same handler?

Comment (zrhoffman) [reply traffic_ops/install/bin/postinstall.py:1354]: `subprocess.SubprocessError` is not defined in Python 2. Separating them like that prevent
```
AttributeError: 'module' object has no attribute 'SubprocessError'
```
in Python 2

File: traffic_ops/install/bin/postinstall.test.sh [hunk 3]
```diff
@@ -15,18 +15,72 @@
...
+while getopts :23sb: opt; do
+	case "$opt" in
+		2)	python_version=2;;
+		3)	python_version=3;;
+		s)	skip_python2=true;;
+		b)	python_bin="$OPTARG";;
+		*)	echo "Invalid flag received: ${OPTARG}" >&2
+			exit 1;;
+	esac;
+done;
```
Comment (ocket8888) [inline traffic_ops/install/bin/postinstall.test.sh:32]: I just realized; this used to have no options. Now that there _are_ options, it'd be nice to have a `-h` and/or `-?` to pass in order to see them.

File: traffic_ops/install/bin/postinstall.py (hunk 2 above)
Comment (ocket8888) [reply traffic_ops/install/bin/postinstall.py:1354]: really? I would expect that to still give an error. I suppose if the test passes that means it's fine.

File: traffic_ops/install/bin/postinstall.py (hunk 1 above)
Comment (zrhoffman) [reply traffic_ops/install/bin/postinstall.py:719]: Passing `e` to the function in d94b755af3

File: traffic_ops/install/bin/postinstall.py [hunk 4]
```diff
@@ -560,48 +703,52 @@ def unmarshal_config(dct: dict) -> Dict[str, List[Question]]:
...
-		subprocess.run(
+		subprocess.check_call(
 			cmd,
 			stderr=subprocess.PIPE,
 			stdout=subprocess.PIPE,
-			check=True,
 			universal_newlines=True
 		)
+	except subprocess.CalledProcessError as e:
+		failed_download(6)
```
Comment (ocket8888) [inline traffic_ops/install/bin/postinstall.py:747]: passing `6` as `Exception` - missing those type hints already? :P

File: traffic_ops/install/bin/postinstall.py [hunk 5]
```diff
@@ -560,48 +703,52 @@ def unmarshal_config(dct: dict) -> Dict[str, List[Question]]:
...
 	except subprocess.SubprocessError as e:
-		logging.error("Failed to download MaxMind data")
-		logging.debug("(ipv6) Exception: %s", e)
+		failed_download(6)
```
Comment (ocket8888) [inline traffic_ops/install/bin/postinstall.py:749]: passing `6` as `Exception`

File: traffic_ops/install/bin/postinstall.py (hunk 4 above)
Comment (zrhoffman) [reply traffic_ops/install/bin/postinstall.py]: PyCharm gave me the warnings, I just didn't scroll down.
Passing 6 as an `int` in ab6881f0fb

File: traffic_ops/install/bin/postinstall.py (hunk 5 above)
Comment (zrhoffman) [reply traffic_ops/install/bin/postinstall.py]: Fixed in ab6881f0fb

File: traffic_ops/install/bin/postinstall.test.sh (hunk 3 above)
Comment (zrhoffman) [reply traffic_ops/install/bin/postinstall.test.sh]: Added help string in 1e7fd5752d

**Review by ocket8888 — APPROVED** [review]
Appears to still work, tests passing.

**Pull request merged**